src/service/fixed-layer.js

```diff
@@ -324,6 +310,44 @@ export class FixedLayer {
   }
 
   /**
+   * Calls `setupFixedSelectors` in a try-catch.
+   * Fails quietly with a dev error if call fails.
+   * This method should not be inlined to prevent TryCatch deoptimization.
+   * NoInline keyword at the end of function name also prevents Closure compiler
+   * from inlining the function.
+   * @param {!Array<string>} fixedSelectors
+   * @private
+   */
+  trySetupFixedSelectorsNoInline(fixedSelectors) {
```

Comment:
Created https://github.com/ampproject/amphtml/issues/4104 to remove this rule when V8 ships TryCatch optimization in Turbofan (which was [shipped](https://codereview.chromium.org/1713033002) and then [unshipped](https://codereview.chromium.org/1931973002) due to regressions it caused)